pkg/controller/gcp/database/instance.go

```diff
@@ -0,0 +1,236 @@
+	// remote instance with this object after some delay interval
+	requeueSync = reconcile.Result{RequeueAfter: requeueAfterSuccess}
+
+	// requeueWait - object processing was partial, due the remote instance is not being ready, i.e.
+	// performing managedOperations like: create  or update. We want to repeat the processing fo this
+	// object after some short(er) (in comparison to above requeueSync) delay
+	requeueWait = reconcile.Result{RequeueAfter: requeueAfterWait}
+
+	log = logging.Logger.WithName("controller." + controllerName)
+)
+
+// Reconciler reconciles cloudsql instance objects
+type Reconciler struct {
+	client.Client
+	factory
+}
+
+// Reconcile reads that state of the cloudsql instance object and makes changes based on the state read
+// and what is in the instance Spec
+func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.V(logging.Debug).Info("reconciling", "kind", v1alpha1.CloudsqlInstanceGroupVersionKind, "request", request)
+
+	ctx, cancel := context.WithTimeout(context.Background(), reconcileTimeout)
+	defer cancel()
+
+	i := &v1alpha1.CloudsqlInstance{}
+	if err := r.Get(ctx, request.NamespacedName, i); err != nil {
+		return requeueNever, handleNotFound(err)
+	}
+
+	// create local operations to handle Kubernetes (local) types operations
+	lops := r.factory.makeLocalOperations(i, r.Client)
+
+	// create managed operations to handle CloudSQL (managed) instance operations
+	mops, err := r.factory.makeManagedOperations(ctx, i, lops)
+	if err != nil {
+		return requeueNow, lops.updateReconcileStatus(ctx, err)
+	}
+
+	// create syncdeleter to handle top level reconciliation operations
+	sd := r.factory.makeSyncDeleter(mops)
+
+	if meta.WasDeleted(i) {
+		return sd.delete(ctx)
+	}
+
+	return sd.sync(ctx)
+}
+
+type factory interface {
+	makeLocalOperations(*v1alpha1.CloudsqlInstance, client.Client) localOperations
+	makeManagedOperations(context.Context, *v1alpha1.CloudsqlInstance, localOperations) (managedOperations, error)
+	makeSyncDeleter(managedOperations) syncdeleter
+}
+
+type operationsFactory struct {
+	client.Client
+}
+
+var _ factory = &operationsFactory{}
+
+func (f *operationsFactory) makeLocalOperations(inst *v1alpha1.CloudsqlInstance, kube client.Client) localOperations {
+	return newLocalHandler(inst, kube)
+}
+
+func (f *operationsFactory) makeManagedOperations(ctx context.Context, inst *v1alpha1.CloudsqlInstance, ops localOperations) (managedOperations, error) {
+	p := &gcpv1alpha1.Provider{}
+	n := meta.NamespacedNameOf(inst.GetProviderReference())
+	if err := f.Get(ctx, n, p); err != nil {
+		return nil, err
+	}
+
+	s := &corev1.Secret{}
+	n = types.NamespacedName{Namespace: p.GetNamespace(), Name: p.Spec.Secret.Name}
+	if err := f.Get(ctx, n, s); err != nil {
+		return nil, errors.Wrapf(err, "cannot get provider's secret %s", n)
+	}
+
+	creds, err := google.CredentialsFromJSON(context.Background(), s.Data[p.Spec.Secret.Key], cloudsql.DefaultScope)
```

Comment:
Is there a reason we make a new context for this call rather than using the `ctx` variable passed into this method?